install/installer/pkg/components/server/configmap.go

```diff
@@ -19,12 +20,49 @@ func configmap(ctx *common.RenderContext) ([]runtime.Object, error) {
 	if err != nil {
 		return nil, err
 	}
+	_ = ctx.WithExperimental(func(cfg *experimental.Config) error {
+		if cfg.WebApp != nil && cfg.WebApp.OAuthServer.JWTSecret != "" {
+			jwtSecret = cfg.WebApp.OAuthServer.JWTSecret
+		}
+		return nil
+	})
 
 	license := ""
 	if ctx.Config.License != nil {
 		license = licenseFilePath
 	}
 
+	workspaceImage := common.ImageName(common.ThirdPartyContainerRepo(ctx.Config.Repository, ""), workspace.DefaultWorkspaceImage, workspace.DefaultWorkspaceImageVersion)
+	_ = ctx.WithExperimental(func(cfg *experimental.Config) error {
+		if cfg.WebApp != nil && cfg.WebApp.WorkspaceDefaults.WorkspaceImage != "" {
+			workspaceImage = cfg.WebApp.WorkspaceDefaults.WorkspaceImage
+		}
+		return nil
+	})
+
+	sessionSecret := "Important!Really-Change-This-Key!"
+	_ = ctx.WithExperimental(func(cfg *experimental.Config) error {
+		if cfg.WebApp != nil && cfg.WebApp.Session.Secret != "" {
+			sessionSecret = cfg.WebApp.Session.Secret
+		}
+		return nil
+	})
+
+	githubApp := GitHubApp{}
+	_ = ctx.WithExperimental(func(cfg *experimental.Config) error {
+		if cfg.WebApp != nil && cfg.WebApp.GithubApp != nil {
```

Comment:
Hmm, perhaps we shouldn't have the config as a pointer but as a value, that'd eliminate some of these extra checks for nil.

Probs not for this PR tho.